src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,173 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class RegistrationServiceTest {
+    private StorageDao storageDao;
+    private RegistrationService registrationService;
+    private User user;
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+    }
+
+    @Test
+    void registerNullUser_NotOk() {
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(null));
+
+        assertEquals("User is null.", exception.getMessage());
+    }
+
+    @Test
+    void registerLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Login can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordNull_NotOk() {
+        user.setLogin("validLogin");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Password can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerUserAgeNull_NotOk() {
+        user.setLogin("validLog");
+        user.setPassword("validPwd");
+        user.setAge(null);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Age can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordExists_NotOk() throws RegistrationException {
+        User newUser = new User();
+        newUser.setLogin("existingUser");
+        newUser.setPassword("password123");
+        newUser.setAge(21);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+    @Test
+    void registerLogin_NotOk() throws RegistrationException {
+        String[] invalidLogins = {"1", "ab", "son", "hell", "user"};
+
+        for (String login : invalidLogins) {
+            user.setLogin(login);
+            user.setPassword("validPassword");
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Login length must be at least 6 characters.", exception.getMessage());
```

Comment:
It's not clear from the task description that there is a minimum login length requirement. If this is not a requirement, remove the test `registerLogin_NotOk`. If it is a requirement, ensure it's documented and tested properly, including edge cases.